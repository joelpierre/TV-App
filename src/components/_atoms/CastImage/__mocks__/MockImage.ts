import { ITvShowImage } from '../../../../types/interfaces';

export const MockImage: ITvShowImage = {
  medium: "http://static.tvmaze.com/uploads/images/medium_portrait/130/325669.jpg",
  original: "http://static.tvmaze.com/uploads/images/original_untouched/130/325669.jpg"
};
