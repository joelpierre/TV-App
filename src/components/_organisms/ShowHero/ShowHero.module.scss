.ShowHero {
  position: relative;
  min-height: 260px;
  padding: spacing(8) 0;

  &::after {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -100;
    background-color: $color-brand;
  }

  @include breakpoint('sm') {
    height: 580px;
    padding: spacing(10) 0;
  }

  &--is-fluid {
    @include breakpoint('sm') {
      height: auto;
    }
  }

  &__container {
    @include max-width-container;
    display: flex;
    flex-wrap: wrap;
  }

  &__content-wrapper {
    display: flex;
    width: 100%;
    margin-bottom: spacing(5);
    flex-direction: column;

    @include breakpoint('sm') {
      flex-direction: row;
    }
  }

  &__image-wrapper {
    width: 100%;
    @include breakpoint('sm') {
      max-width: 272px;
    }
  }

  &__content {
    display: flex;
    flex-direction: column;
    width: 100%;
    padding-top: spacing(4);

    @include breakpoint('sm') {
      width: 60%;
      padding-left: spacing(6);
    }
  }

  &__link {
    margin-top: -#{spacing(3)};
    margin-bottom: spacing(5);
    color: $color-tint-alpha;
    text-decoration: none;
    padding-left: 0;
    transition: padding-left .4s;

    @include hover-focus {
      text-decoration: none;
      color: $color-tint-alpha;
      padding-left: spacing();
    }
  }

  &__heading {
    @include type-heading-1;
    margin-bottom: spacing(7);
    color: $color-tint-alpha;
    font-size: 28px;
    line-height: 1;
    width: 100%;
  }

  &__ratings {
    margin-bottom: spacing(4);
    color: $color-tint-alpha;
    font-size: 20px;
  }

  &__show-title {
    font-size: 50px;
    width: 100%;
    margin-bottom: spacing(4);
    color: $color-tint-alpha;
    font-family: $font-regular;
  }

  &__copy {
    $clamp: 2;
    font-size: 20px;
    line-height: 1.5;
    color: $color-tint-alpha;

    &--is-clamped {
      display: box;
      overflow: hidden;
      -webkit-box-orient: vertical;
      display: -webkit-box;
      line-clamp: $clamp;
      -webkit-line-clamp: $clamp;
    }
  }

  &__read-more {
    width: 140px;
    display: inline;
    appearance: none;
    border: none;
    background-color: transparent;
    text-align: left;
    color: $color-tint-alpha;
    font-size: 20px;
    margin-top: spacing(2);
    font-family: $font-semibold;
    text-decoration: underline;
    cursor: pointer;

    @include hover-focus {
      text-decoration: none;
    }
  }
}
