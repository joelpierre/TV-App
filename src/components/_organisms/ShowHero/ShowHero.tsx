import classNames from 'classnames';
import React from 'react';

import styles from './ShowHero.module.scss';
import { ITvShow } from 'common/types/interfaces';
import ShowImage from '@atoms/ShowImage';
import Ratings from '@molecules/Ratings';
import Anchor from '@atoms/Anchor';
import { EPageHref } from 'common/types/enums';
import useToggle from '@hooks/useToggle';

interface IShowHeroProps {
  className?: string;
  show: ITvShow;
}

const ShowHero: React.FunctionComponent<IShowHeroProps> = ({
  className,
  show: { image, name, rating: { average }, summary }
}) => {
  const [isOpen, setIsOpen] = useToggle(false);

  return (
    <article className={classNames(styles.ShowHero, className, {
      [styles['ShowHero--is-fluid']]: isOpen
    })}>
      <div className={styles.ShowHero__container}>
        <Anchor className={styles.ShowHero__link} href={EPageHref.Home} as="/">
          {'<'} Back to Schedule
        </Anchor>
        <span className={styles.ShowHero__heading}>
          TV Bland
        </span>

        <div className={styles['ShowHero__content-wrapper']}>
          <div className={styles['ShowHero__image-wrapper']}>
            <ShowImage image={image} name={name} />
          </div>

          <div className={styles.ShowHero__content}>
            <Ratings averageRating={average} className={styles.ShowHero__ratings} withFigures={true} />
            <h1 className={styles['ShowHero__show-title']}>
              {name}
            </h1>
            <div
              className={classNames(styles.ShowHero__copy, {
                [styles['ShowHero__copy--is-clamped']]: !isOpen
              })}
              dangerouslySetInnerHTML={{ __html: summary }}
            />
            <button className={styles['ShowHero__read-more']} onClick={setIsOpen}>
              {isOpen ? 'Read less' : 'Read more'}
            </button>
          </div>
        </div>
      </div>
    </article>
  );
};

export default ShowHero;
