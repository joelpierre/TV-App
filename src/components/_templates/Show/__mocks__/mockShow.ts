/* eslint-disable */
import { ITvShow } from '../../../../types/interfaces';
import { MockImage } from '../../../_atoms/CastImage/__mocks__/MockImage';

export const MockShow: ITvShow = {
  "id": 31534,
  "url": "http://www.tvmaze.com/shows/31534/the-russell-howard-hour",
  "name": "The Russell Howard Hour",
  "type": "Talk Show",
  "language": "English",
  "genres": [
    "Comedy"
  ],
  "status": "Running",
  "runtime": 60,
  "premiered": "2017-09-21",
  "officialSite": "https://www.sky.com/watch/title/series/89419c88-0079-426a-b29b-bc28bc403a58/the-russell-howard-hour",
  "schedule": {
    "time": "22:00",
    "days": [
      "Thursday"
    ]
  },
  "rating": {
    "average": 8.4
  },
  "weight": 83,
  "network": {
    "id": 63,
    "name": "Sky 1",
    "country": {
      "name": "United Kingdom",
      "code": "GB",
      "timezone": "Europe/London"
    }
  },
  "webChannel": null,
  "externals": {
    "tvrage": null,
    "thetvdb": 334902,
    "imdb": "tt7544016"
  },
  "image": MockImage,
  "summary": "<p><b>The Russell Howard Hour</b> is a weekly talk show in which Russell Howard will be chatting with celebrity guests alongside comedy stand-up. </p>",
  "updated": 1603404686,
  "_links": {
    "self": {
      "href": "http://api.tvmaze.com/shows/31534"
    },
    "previousepisode": {
      "href": "http://api.tvmaze.com/episodes/1955660"
    },
    "nextepisode": {
      "href": "http://api.tvmaze.com/episodes/1955661"
    }
  },
  "_embedded": {
    "cast": [
      {
        "person": {
          "id": 64866,
          "url": "http://www.tvmaze.com/people/64866/russell-howard",
          "name": "Russell Howard",
          "country": {
            "name": "United Kingdom",
            "code": "GB",
            "timezone": "Europe/London"
          },
          "birthday": "1980-03-23",
          "deathday": null,
          "gender": "Male",
          "image": {
            "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/175/437607.jpg",
            "original": "http://static.tvmaze.com/uploads/images/original_untouched/175/437607.jpg"
          },
          "_links": {
            "self": {
              "href": "http://api.tvmaze.com/people/64866"
            }
          }
        },
        "character": {
          "id": 456592,
          "url": "http://www.tvmaze.com/characters/456592/the-russell-howard-hour-host",
          "name": "Host",
          "image": null,
          "_links": {
            "self": {
              "href": "http://api.tvmaze.com/characters/456592"
            }
          }
        },
        "self": true,
        "voice": false
      }
    ]
  }
};
