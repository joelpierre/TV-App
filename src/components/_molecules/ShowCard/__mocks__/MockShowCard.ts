import { ITvSchedule } from '../../../../types/interfaces';
import { MockShow } from '../../../_templates/Show/__mocks__/mockShow';
import { MockImage } from '../../../_atoms/CastImage/__mocks__/MockImage';

export const MockShowCard: ITvSchedule = {
  url: '',
  type: '',
  summary: '',
  season: 2,
  runtime: 50,
  number: 2,
  name: 'Test Show Card',
  image: MockImage.original,
  id: 123,
  airtime: '09:00',
  airstamp: 'airStamp',
  airdate: '01-01-2020',
  show: MockShow
};
