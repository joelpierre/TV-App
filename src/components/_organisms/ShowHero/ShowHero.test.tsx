import React from 'react';
import { fireEvent, render, RenderResult, screen } from 'test-utils';
import ShowHero, { IShowHeroProps } from './ShowHero';
import { MockShow } from '../../_templates/Show/__mocks__/mockShow';

const setup = (props: Partial<IShowHeroProps> = {}): RenderResult => {
  const setupProps = { show: MockShow, ...props };
  return render(<ShowHero {...setupProps} />);
};

describe('<ShowHero/>', () => {
  let wrapper: RenderResult;

  // Default State
  describe('Given isOpen is false', () => {
    beforeEach(() => {
      wrapper = setup();
    });

    it('Should render without crashing', () => {
      expect(wrapper.container).not.toBeEmptyDOMElement();
    });

    it('should render a H1 tag', () => {
      expect(wrapper.container.getElementsByTagName('h1')).toHaveLength(1);
    });

    it('should initially render with a read-more button', () => {
      expect(wrapper.getByTestId('read-more')).toBeTruthy();
      expect(wrapper.queryByTestId('read-less')).toBeFalsy();
    });

    it('should match snapshot', () => {
      expect(wrapper.asFragment()).toMatchSnapshot();
    });

    it('should toggle read-more', () => {
      const readMoreBtn = wrapper.getByTestId('read-more');

      fireEvent.click(readMoreBtn);

      expect(screen.getByText('Read less')).toBeTruthy();
      expect(wrapper.queryByTestId('read-more')).toBeFalsy();
    });
  });

  describe('Given isOpen is true', () => {
    beforeEach(() => {
      wrapper = setup();
      const readMoreBtn = wrapper.getByTestId('read-more');
      fireEvent.click(readMoreBtn);
    });

    it('Should render without crashing', () => {
      expect(wrapper.container).not.toBeEmptyDOMElement();
    });

    it('should render with a read-less button', () => {
      expect(wrapper.getByTestId('read-less')).toBeTruthy();
      expect(wrapper.queryByTestId('read-more')).toBeFalsy();
    });

    it('should toggle read-less', () => {
      const readLessBtn = wrapper.getByTestId('read-less');
      fireEvent.click(readLessBtn);
      expect(screen.getByText('Read more')).toBeTruthy();
      expect(wrapper.queryByTestId('read-less')).toBeFalsy();
    });
  });
});
