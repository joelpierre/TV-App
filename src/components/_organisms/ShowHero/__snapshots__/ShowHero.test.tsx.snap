// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowHero/> Given isOpen is false should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="ShowHero"
  >
    <div
      class="ShowHero__container"
    >
      <a
        class="Anchor ShowHero__link"
        href="/"
      >
        &lt; Back to Schedule
      </a>
      <span
        class="ShowHero__heading"
      >
        TV Bland
      </span>
      <div
        class="ShowHero__content-wrapper"
      >
        <div
          class="ShowHero__image-wrapper"
        >
          <div
            class="ShowImage"
          >
            <img
              alt="The Russell Howard Hour poster"
              class="ShowImage__image"
              loading="lazy"
              src="http://static.tvmaze.com/uploads/images/medium_portrait/130/325669.jpg"
              width="200"
            />
          </div>
        </div>
        <div
          class="ShowHero__content"
        >
          <div
            class="Ratings ShowHero__ratings"
          >
            <span
              class="Star Ratings__star Star--is-active"
            />
            <span
              class="Star Ratings__star Star--is-active"
            />
            <span
              class="Star Ratings__star Star--is-active"
            />
            <span
              class="Star Ratings__star Star--is-active"
            />
            <span
              class="Star Ratings__star"
            />
            <span
              class="Ratings__figures"
              data-testid="figures"
            >
              4.2/5
            </span>
          </div>
          <h1
            class="ShowHero__show-title"
          >
            The Russell Howard Hour
          </h1>
          <div
            class="ShowHero__copy ShowHero__copy--is-clamped"
          >
            <p>
              <b>
                The Russell Howard Hour
              </b>
               is a weekly talk show in which Russell Howard will be chatting with celebrity guests alongside comedy stand-up. 
            </p>
          </div>
          <button
            class="ShowHero__read-more"
            data-testid="read-more"
          >
            Read more
          </button>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
